src/resdk/tables/variant.py

```diff
@@ -0,0 +1,424 @@
+            # the above regex can't parse it
+            warnings.warn(f'Cannot encode mutation from value "{allele_line}".')
+            return np.nan
+
+        if allele1 == allele2 == row["REF"]:
+            return 0
+        elif allele1 == allele2 == row["ALT"]:
+            return 2
+        else:
+            return 1
+
+    def _parse_file(self, file_obj, sample_id, data_type) -> pd.Series:
+        """Parse file - get encoded variants / depth of a single sample."""
+        sample_data = pd.read_csv(file_obj, sep="\t")
+
+        # Filter mutations if specified
+        if self.mutations:
+            keep = set()
+            for gene, aa_change in self.parse_mutations_string(self.mutations).items():
+                for sample, gene2, aa_change2 in zip(
+                    sample_data.index, sample_data["Gene_Name"], sample_data["HGVS.p"]
+                ):
+                    if not aa_change:
+                        if gene == gene2:
+                            keep.add(sample)
+                    else:
+                        if gene == gene2 and aa_change == self.strip_amino_acid(
+                            aa_change2, remove_alt=True
+                        ):
+                            keep.add(sample)
+
+            sample_data = sample_data.loc[keep]
+
+        # Construct index
+        sample_data["index"] = sample_data.apply(
+            self.construct_index, axis=1, result_type="reduce"
+        )
+        sample_data.set_index("index", inplace=True)
+        sample_data.index.name = None
+
+        if data_type == self.VARIANTS:
+            s = sample_data.apply(self.encode_mutation, axis=1, result_type="reduce")
+        elif data_type == self.DEPTH:
+            # Depth, as computed by GATK is reported by "DP" column
+            s = sample_data["Total_depth"]
+        elif data_type == self.DEPTH_A:
+            s = sample_data["Base_A"]
+        elif data_type == self.DEPTH_C:
+            s = sample_data["Base_C"]
+        elif data_type == self.DEPTH_G:
+            s = sample_data["Base_G"]
+        elif data_type == self.DEPTH_T:
+            s = sample_data["Base_T"]
+
+        s.name = sample_id
+        return s
+
+    def postprocess_variants_table(self, df, data_type):
+        """
+        Propagate info from "fake" to the "real" variants.
+
+        "fake variants" are referred as positions for which we know that
+        do not contain variants. The absence of variants is still a valuable
+        information and should be propagated further when mergeing info
+        from multiple samples in matrix form.
+        """
+
+        def chrom_pos(column_name):
+            """Return only chromosome and position from index."""
+            chrom, pos = column_name.split("_")[:2]
+            return f"{chrom}_{pos}"
+
+        to_discard = list()
+
+        for _, group in groupby(df.columns, key=chrom_pos):
+            # Sorting moves fake variant to the first position
```

Comment:
There is an issue if you first call `VariantTables` with `discard_fakes` set to default (False) and then try to call `VariantTables` again but with `discard_fakes` set to True. "fakes" will not be removed in the second call, you first have to clear the cache. The default value for `discard_fakes` could be set to True.